File: 백준/깊이 우선 탐색/2606.py
v = int(input())
e = int(input())
dic = {}
visited = []

def dfs(now, dic):
  for i in dic[now]:
    if i not in visited:
      visited.append(i)
      dfs(i, dic)


for i in range(1,v+1):
  dic[i] = []
for i in range(e):
  a,b = map(int, input().split())
  dic[a].append(b)
  dic[b].append(a)
dfs(1, dic)
print(len(visited)-1)

File: 백준/dfs_bfs/s3/2606.js
const fs = require("fs");
let [n, count, ...arr] = fs.readFileSync("./data", "utf8").trim().split("\n");

const stored = [...new Array(parseInt(n, 10) + 1)].map(() => []);
const visited = [...new Array(parseInt(n, 10) + 1)].map(() => false);

visited[1] = true;
let ans = 0;

arr.map((el) => {
  const [from, to] = el.split(" ").map((v) => +v);
  stored[from].push(to);
  stored[to].push(from);
});

const dfs = (from) => {
  stored[from].forEach((to) => {
    if (!visited[to]) {
      visited[to] = true;
      ans += 1;
      dfs(to);
    }
  });
};

dfs(1);
console.log(ans);
